src/TopoMojo.Hypervisor/Proxmox/Models/PveNodeTask.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+using System.Xml;
+using Corsinvest.ProxmoxVE.Api;
+
+namespace TopoMojo.Hypervisor.Proxmox.Models
+{
+    internal class PveNodeTask
+    {
+        public string Id { get; set; }
+        public string Action { get; set; }
+        public int Progress { get; private set; }
+
+        // The Line Number that was last processed to get the current Progress
+        public int? LastLine { get; private set; } = null;
+
+        public DateTimeOffset WhenCreated { get; set; }
+
+        public void SetProgress(PveNodeTaskLog log)
+        {
+            Progress = log.Progress;
```

Comment:
I'm honestly not sure if this matters because I don't think I have the _entire_ picture in my head, but I would maybe not set the public value of `Progress until you've clamped it to be under 100 (below), because you do examine that value and pass it to Topo on 496 of the ProxmoxClient. Yes, this is insanely nitpicky.

Reply:
Good catch, thanks